### PR TITLE
hunspell: add missing dictionaries

### PR DESCRIPTION
**Pull request by 89luca89**
Related: https://github.com/chainguard-dev/image-requests/issues/3637

Fixing subpackages creation for dictinaries, needed for spellchecking to actually work

### CODE REVIEW

File: hunspell.yaml [hunk 1]
```diff
@@ -32,6 +33,26 @@ pipeline:
 
   - uses: strip
 
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/share/hunspell
+      cd ${{targets.destdir}}/usr/share/hunspell
+      wget -c https://downloads.sourceforge.net/project/wordlist/speller/2020.12.07/hunspell-en_US-2020.12.07.zip
+      wget -c https://downloads.sourceforge.net/project/wordlist/speller/2020.12.07/hunspell-en_CA-2020.12.07.zip
+      wget -c https://downloads.sourceforge.net/project/wordlist/speller/2020.12.07/hunspell-en_AU-2020.12.07.zip
+      wget -c https://downloads.sourceforge.net/project/wordlist/speller/2020.12.07/hunspell-en_GB-large-2020.12.07.zip
```
Comment (rawlingsj) [inline hunspell.yaml:42]: I think we'd want to find a better way to fetch these, are these available via a git checkout?  We'd maybe want to split these into subpackages using a data / range feature of melange similar to https://github.com/wolfi-dev/os/blob/120fd8a3abda7a5c729907410d683481505cf2d3/glibc.yaml#L126-L131 and https://github.com/wolfi-dev/os/blob/120fd8a3abda7a5c729907410d683481505cf2d3/glibc.yaml#L383-L389

also this change on the package would require an epoch bump for new apks to include the changes

Comment (89luca89) [reply hunspell.yaml]: Thanks for the feedback @rawlingsj  :)
Found a better source of dicts from libreoffice, so I'm now using git for that

Split package is now done with data/range over `_dictionaries` :+1:

File: hunspell.yaml [hunk 2]
```diff
@@ -49,11 +49,6 @@ subpackages:
       - uses: split/manpages
     description: hunspell manpages
 
-  - name: hunspell-lang
```
Comment (dlorenc) [inline hunspell.yaml:52]: Should we withdraw this package too?

File: hunspell-dict.yaml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      sw:
+      te:
+      th:
+      tr:
+      uk:
+      vi:
+
+subpackages:
+  - range: _dictionaries
+    name: hunspell-dictionary-${{range.key}}
```
Comment (dlorenc) [inline hunspell-dict.yaml:154]: Just curious - some other distros seem to call these packages "hunspell-$lang" without the -dictionary- part, did you copy that convention from somewhere?

**Comment by dlorenc** [comment]
Looks good, just two comments.

File: hunspell.yaml (hunk 2 above)
Comment (89luca89) [reply hunspell.yaml:52]: It was empty anyway, hunspell doesn't provide locales

File: hunspell-dict.yaml (hunk 3 above)
Comment (89luca89) [reply hunspell-dict.yaml:154]: usually `-lang` is for locales (like `glibc-lang-foo`) these are dictionaries, so I didn't want to make it confusing

on alpine they just use `hunspell-en-...` without `-dictionaries` but I thought having a separate package like this enables us also to have an `-all` package that depends on the subpackages

This enables apk to properly handle adding `-all` after a subpackage without overwriting files

**Review by dlorenc — APPROVED** [review]

**Pull request merged**